pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -335,7 +338,7 @@ self: super: {
     then dontCheck super.math-functions # "erf table" test fails on Darwin https://github.com/bos/math-functions/issues/63
     else super.math-functions;
   matplotlib = dontCheck super.matplotlib;
-
+  matterhorn = doJailbreak super.matterhorn; # this is needed until the end of time
```

Comment:
> this is needed until the end of time

Is there some reason why this can't be reported upstream and get the author to fix the bounds in the .cabal file?

For instance, when trying to build `matterhorn` without `doJailbreak`, Cabal complains about:

```console
$ nix-build -A haskellPackages.matterhorn
...
configuring matterhorn-50200.11.0...
CallStack (from HasCallStack):
  $, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:1024:20 in Cabal-3.2.0.0:Distribution.Simple.Configure
  configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:477:12 in Cabal-3.2.0.0:Distribution.Simple.Configure
  configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:625:20 in Cabal-3.2.0.0:Distribution.Simple
  confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:65:5 in Cabal-3.2.0.0:Distribution.Simple.UserHooks
  configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:180:19 in Cabal-3.2.0.0:Distribution.Simple
  defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:116:27 in Cabal-3.2.0.0:Distribution.Simple
  defaultMain, called at Setup.hs:2:8 in main:Main
Setup: Encountered missing or private dependencies:
aeson >=1.2.3.0 && <1.5,
aspell-pipe ==0.4.*,
skylighting-core ==0.8.*,
strict ==0.3.*,
vty ==5.31.*
```

You can check which versions of these things we have in nixpkgs:

```console
$ nix-broken -A haskellPackages.aeson
/nix/store/kdlv4s53whzllsvv5xqnn15f6lv9z7zi-aeson-1.5.4.1
```

So the `matterhorn.cabal` file would need the `aeson` version bound bumped to probably `aeson  < 1.6`.  The other dependencies should have similar bumps made.

It looks like there are other PRs making these bumps, so the author is probably okay with these types of things: https://github.com/matterhorn-chat/matterhorn/pull/487

Reply:
lol yes, because I was being funny and didn't have an issue to put there yet. I have too many things going on and get distracted by shiny things. I was in the mood to update the PR with the changes for glirc before I was in the mood to file an issue; I hate making issues. "No Issues! only PRs!" I have made one now

Reply:
@Kiwi could you update the code here to link to the issue so that the nixpkg Haskell maintainers can track it?

Reply:
yeah I am about to do that now